### ISSUE TITLE
StartExecution of statemachines through API Gateway

### ISSUE THREAD

**Issue by msl-kabo** [horike37/serverless-step-functions#25]
It would be really good to have the ability to start execution of a statemachine by calling a url.

https://docs.aws.amazon.com/step-functions/latest/dg/tutorial-api-gateway.html

Suggestion for how it could look in `serverless.yml`

```yaml
stepFunctions:
  stateMachines:
    myStateMachine:
      events:
        - http:
          method: GET
          path: executeMyStateMachine
```

**Comment by horike37** [comment]
Thank you for the suggestion @msl-kabo !
will do it.

**Comment by ShaunEgan** [comment]
Is there any movement on `0.5.0`?

**Comment by horike37** [comment]
@ShaunEgan 
Thank you for the pinging!
The plan is not set yet.

I am considering sending a PR to be included into the core at this timing
https://github.com/serverless/serverless/issues/3024

**Comment by ShaunEgan** [comment]
Out of curiosity: 

I read discussion that `step-functions` might not make it into core since they are too AWS specific? Would it not be better to get them in here and keep this the defacto plugin for step functions? 

I would love to have this functionality available here, I am currently building a project which leans on this plugin and would love to have one place to build my step functions and gateway for them. 

Either way, I am happy.

**Comment by ezjacob-maheim** [comment]
is there a way around this till this feature is supported? I'm building something serverless and would love to be able to trigger the state machine from API Gateway without manually doing it.

**Comment by ShaunEgan** [comment]
Not without building it through cloudformation or doing it manually. As far as I am aware, this is coming really soon.

**Comment by horike37** [comment]
Thank you for the comment @ezjacob-maheim and helping @ShaunEgan :smile:
Actually the feature will be supported on next release(v1.0). Then I have finished to implement on 
the head of master branch.

If you need this feature before next release, you can build Lambda function to trigger a statemachine.

**Comment by ShaunEgan** [comment]
This should be closed by #33 now 😄

**Comment by horike37** [comment]
@ShaunEgan 
Thanks for the comment. yes, closed now :smile:
